lib/internal/bootstrap/node.js

```diff
@@ -611,6 +611,21 @@
       // It's possible that defaultTriggerAsyncId was set for a constructor
       // call that threw and was never cleared. So clear it now.
       clearDefaultTriggerAsyncId();
+      try {
+        // if node-report is enabled, call into its handler to see
+        // whether it is interested in handling the situation.
+        const report = internalBinding('node_report');
+        if (report !== undefined && report !== null) {
+          if (process.binding('config').experimentalReport) {
+            const config = {};
+            report.syncConfig(config, false);
+            if (config.events !== undefined &&
+                config.events.includes('exception')) {
+              process.emit('reportuncaught', er.stack);
```

Comment:
Is it intentional that this runs even if a `domain` catches the exception?

Also, I’d prefer to avoid using a publicly accessible event for this … if you want to go with `process.emit()` here, can you use a Symbol?

Reply:
Oh, also, maybe relevant: The discussion at https://github.com/nodejs/node/pull/17159#issuecomment-427208024 about supporting multiple capture callbacks

Reply:
Excepted the domain cases.
I have devised the emitting logic because I didn't want a coupling between `node.js` and `util.js` (needing to require util module in the bootstrap). But between the development process, I forgot that I am syncing the config from the C++ land, your this comment reminded me of that. So now I can directly call OnUnCaughtException if all conditions are met, avoiding needing to emit, and usage of symbols.
re: supporting multiple capture callbacks: I guess as we neither absorbing the exception nor re-throwing, we are not changing the behavior in any ways, rather just capturing a report in between; I can revisit this in future in the context of 
https://github.com/nodejs/node/pull/17159#issuecomment-427208024